Common/Data/Custom/Quiver/QuiverWallStreetBets.cs

```diff
@@ -0,0 +1,133 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Util;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using NodaTime;
+using ProtoBuf;
+using static QuantConnect.StringExtensions;
+
+namespace QuantConnect.Data.Custom.Quiver
+{
+    /// <summary>
+    /// Mentions of the given company's ticker in the WallStreetBets daily discussion thread
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public class QuiverWallStreetBets : BaseData
+    {
+        /// <summary>
+        /// Date of the daily discussion thread
+        /// </summary>
+        [ProtoMember(10)]
+        [JsonProperty(PropertyName = "Date")]
+        [JsonConverter(typeof(DateTimeJsonConverter), "yyyy-MM-dd")]
+        public DateTime Date { get; set; }
+
+        /// <summary>
+        /// The number of mentions on the given date
+        /// </summary>
+        [ProtoMember(11)]
+        [JsonProperty(PropertyName = "Count")]
+        public int Mentions { get; set; }
+
+        /// <summary>
+        /// Required for successful Json.NET deserialization
+        /// </summary>
+        public QuiverWallStreetBets()
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of QuiverWallStreetBets from a CSV line
+        /// </summary>
+        /// <param name="csvLine">CSV line</param>
+        public QuiverWallStreetBets(string csvLine)
+        {
+            // Date[0], Mentions[1]
+            var csv = csvLine.Split(',');
+            Date = Parse.DateTimeExact(csv[0], "yyyyMMdd");
+            Mentions = Parse.Int(csv[1]);
+            Time = Date;
+        }
+
+        /// <summary>
+        /// Return the Subscription Data Source gained from the URL
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>Subscription Data Source.</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            var source = Path.Combine(
```

Comment:
For live trading all these new data sources should throw an InvalidOperationException?